python/sglang/lang/ir.py

```diff
@@ -461,14 +462,22 @@ def __repr__(self):
 
 
 class SglSelect(SglExpr):
-    def __init__(self, name: str, choices: List[str], temperature: float):
+
+    def __init__(
+        self,
+        name: str,
+        choices: List[str],
+        temperature: float,
+        choices_method: Optional[ChoicesSamplingMethod],
```

Comment:
Do not make it optional.
The IR here is expected to be fully deterministic across versions and platforms, so please do not leave any ambiguous optional arguments here.

We should resolve the case of `choices_method` being `None` at the very beginning, such as in `api.py`.